src/spac/spatial_analysis.py

```diff
@@ -838,3 +839,109 @@ def _neighborhood_profile_core(
         neighborhood_array = neighborhood_array / bins_areas[np.newaxis, np.newaxis, :]
 
     return neighborhood_array
+
+
+def calculate_spatial_distance(
+    adata,
+    annotation,
+    obsm_key='spatial',
+    subset=None,
+    imageid='imageid',
+    label='spatial_distance',
+    verbose=True
+):
+    """
+    Calculates spatial distances and stores the results in `adata.uns`.
+
+    Parameters:
+        adata (AnnData): Annotated data matrix with spatial information.
+        obsm_key (str): Key in `adata.obsm` where spatial coordinates are stored.
+        annotation (str): Column name in `adata.obs` containing cell annotations.
+        subset (str or list, optional): Subset of image IDs to process.
+        imageid (str): Column name in `adata.obs` containing image identifiers.
+        label (str): Key under which to store the result in `adata.uns`.
+        verbose (bool): If True, prints progress messages.
+
+    Returns:
+        AnnData: The input `adata` object with updated `adata.uns[label]`.

```

Comment:
@fangliu117 , it is not clear to me what would be stored in adata.uns[label]? Would you please elaborate in the description and probably add an example?

Reply:
According to CR, I have updated the docstring by adding example.
Returns
    -------
    None
        This function modifies `adata` in place by storing the calculated
        spatial distance matrix in `adata.uns[label]`.

    Notes
    -----
    The results saved in `adata.uns[label]` include a dictionary with
    the following structure:

    - 'distance_matrix': A NumPy array of pairwise spatial distances.
    - 'phenotype': The annotation used for grouping cells.
    - 'image_subset': The subset of images processed, if specified.

    Example
    -------
    >>> calculate_spatial_distance(
    ...     adata, annotation='cell_type', obsm_key='spatial',
    ...     subset=['Image1', 'Image2'], imageid='image_id', label='spatial_dist'
    ... )
    >>> print(adata.uns['spatial_dist'])
    {
        'distance_matrix': array([[0.0, 1.2, ...], [...]]),
        'phenotype': 'cell_type',
        'image_subset': ['Image1', 'Image2']
    }

Also, add an unit test "test_output_with_different_setups"
- Hardcoded expectations for easy understanding of what is stored in adata.uns['spatial_distance'].
- Three scenarios: one slide/one phenotype, one slide/two phenotypes, and two slides/one phenotype.
@georgezakinih